dags/hivemind_etl_helpers/discord_mongo_summary_etl.py

```diff
@@ -0,0 +1,122 @@
+import argparse
+import logging
+
+from llama_index import Document
+from llama_index.response_synthesizers import get_response_synthesizer
+
+from hivemind_etl_helpers.src.db.discord.discord_summary import DiscordSummary
+from hivemind_etl_helpers.src.db.discord.find_guild_id import (
+    find_guild_id_by_community_id,
+)
+from hivemind_etl_helpers.src.document_node_parser import configure_node_parser
+from hivemind_etl_helpers.src.utils.cohere_embedding import CohereEmbedding
+from hivemind_etl_helpers.src.utils.pg_db_utils import setup_db
+from hivemind_etl_helpers.src.utils.pg_vector_access import PGVectorAccess
+
+
+def process_discord_summaries(
+    community_id: str, verbose: bool = False
+) -> list[Document]:
+    """
+    prepare the discord data by grouping it into thread, channel and day
+    Note: This will always process the data until 1 day ago.
+
+    Parameters
+    ------------
+    community_id : str
+        the community id to process its guild data
+    verbose : bool
+        verbose the process of summarization or not
+        if `True` the summarization process will be printed out
+        default is `False`
+
+    Returns
+    ---------
+    messages_docuemnt : list[llama_index.Document]
+        list of messages converted to documents
+    """
+    guild_id = find_guild_id_by_community_id(community_id)
+    logging.info(f"COMMUNITYID: {community_id}, GUILDID: {guild_id}")
+    table_name = "discord_summary"
+    dbname = f"community_{community_id}"
+
+    latest_date_query = f"""
+            SELECT (metadata_->> 'date')::timestamp
+            AS latest_date
+            FROM data_{table_name}
+            ORDER BY (metadata_->>'date')::timestamp DESC
+            LIMIT 1;
+    """
+    from_date = setup_db(
+        community_id=community_id, dbname=dbname, latest_date_query=latest_date_query
+    )
+
+    discord_summary = DiscordSummary(
+        response_synthesizer=get_response_synthesizer(response_mode="tree_summarize"),
+        verbose=verbose,
+    )
+
+    (
+        thread_summaries_documents,
+        channel_summary_documenets,
+        daily_summary_documenets,
+    ) = discord_summary.prepare_summaries(
+        guild_id=guild_id,
+        from_date=from_date,
+        summarization_query="Please give me a summary using the data you have!",
```

Comment:
I would suggest this query:

"Please make a concise summary based only on the provided text from this {text_type}"

text_type = "discord thread", "selection of discord thread summaries", or "selection of discord channel summaries"

I'm assuming that below this line, the actual messages are added.

Reply:
Using this, it would uses the `llama_index.Document`s to get the summary and it is not based on the text to be added under the summarization query.
Anyway, I made the changes for each thread, channel, and day summaris.

Reply:
I'm not sure if I understand. Are you saying that the default prompt from LlamaIndex is used? If so, why is a different summarization_query defined if it's not used?

Reply:
There's no default prompt as far as I know in llama_index library. To refine my answer I can say
we're using the created documents in llama_index that do have the metadata as an addition to just the text to make a summary, So it's not just the string to be sent. This answer was for the part of your message
> I'm assuming that below this line, the actual messages are added.

Reply:
Thanks for clarifying. Good to know we're using our own prompt here. When making the summaries, I think we should not include the metadata and instead only use the message text. The metadata is used for optionally filtering a search and for providing the source of the final answer.

Reply:
I'm following this instruction to create summaries with out own prompts (as you suggested).
https://docs.llamaindex.ai/en/stable/examples/index_structs/doc_summary/DocSummary.html#build-document-summary-index

I can remove the metadata from the documents when adding them to summarize but I think having them could make the summarization process better (LLM would have more knowledge as date, replies, mentions, etc of the message). 
If you still think I have to remove the metadata from documents I'll do so. Please let me know.

Reply:
Great, the linked example is perfect. Although the example doesn't include any metadata. For our case, some metadata is redundant since all messages have the same date and channel/thread. Furthermore, since one of the main uses of the summaries is to get the most important discussed topics (see card 3b and 3c), adding info about replies/mentions/reactions is not helpful. These are all distractions for GPT that could leak into the summary and make it less about the actual discussed topics and more about who said what and where. Therefore, I think it's best to leave the metadata out in this case.

Reply:
Sure, thanks then. I'll update the codes to exclude the metadata. Before doing that, I'm asking my final question which is. Do we need to exclude all metadata or just keep some of them which I listed below?
- date
- author_username
- author_nickname
- author_global_name

Reply:
I think at least having the `date` could be useful. What do you think?

Reply:
I don't think date will be useful since we're only summarizing messages over the same date in this stage. And so the date will be the same for all messages. The author name might be nice but that would make it difficult to anonymize names in the future (this is a feature we've promised). So for now, I think it would be best not to include any metadata.

Reply:
Sure then, thanks. I have already updated the codes today to not to include the metadata for summaries. I'll notify you after finishing up the CIs on this branch.